File: atividade-5/atividade.py
class Animal:
    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        return "Latido"

class Gato(Animal):
    def som(self):
        return "Miau"

def emitir_sons(animais):
    for animal in animais:
        print(animal.som())

animais = [Cachorro(), Gato()]
emitir_sons(animais)

File: atividade-5/atividade.go
package main

import "fmt"

type Animal interface {
	Som() string
}

type Cachorro struct{}
type Gato struct{}

func (c Cachorro) Som() string {
	return "Latido"
}

func (g Gato) Som() string {
	return "Miau"
}

func emitirSons(animais []Animal) {
	for _, animal := range animais {
		fmt.Println(animal.Som())
	}
}

func main() {
	animais := []Animal{Cachorro{}, Gato{}}
	emitirSons(animais)
}
